### DISCUSSION TITLE
How to get multiple lines on HTML

### DISCUSSION THREAD

**Discussion by hakushun** [remarkjs/remark#876] [Q&A] [answered]
Could you please tell me how to get multiple lines on HTML.

I know that this issue has already discussed before, but I can't get multiple lines so create this issue.

## my dev env is
- Windows 10
- node: v16.9.0
- next: 11.1.2
- react: 17.0.2
- react-markdown :7.0.1
- remark-breaks: 3.0.1
- remark-gfm: 2.0.0

## This is my code.
I’d like to reflect multiple breaks between 'ssssssssssssssss' and ' # h1' because I create a text form which can input both plain text and markdown.
```
import type { VFC } from "react";
import ReactMarkdown from "react-markdown";
import remarkGfm from "remark-gfm";
import breaks from "remark-breaks";
import { PrismAsyncLight as SyntaxHighlighter } from "react-syntax-highlighter";
import { vscDarkPlus } from "react-syntax-highlighter/dist/esm/styles/prism";
import { CodeComponent } from "react-markdown/lib/ast-to-react";
import "github-markdown-css/github-markdown.css";

const markdown = `
  aaa      aa   a


  ssssssssssssssss




  # h1
  ## h2
  ### h3
  #### h4
  ##### h5
  ###### h6

  Just a link: https://reactjs.com.

  [google](https://www.google.co.jp/)へのLink

  **1234567890**

  ~1234567890~

  _1234567890_

  \`code\`

  ## List
  - a
  - b
  - c

  ## Order
  1. a
  1. b
  1. c

  ## Table
  |left | center | right|
  |:--- | :---:  |  ---:|
  | 1   |2       | 3    |
  | 1   |2       | 3    |
  | 1   |2       | 3    |

  \`\`\`tsx
    const a = apple;
    const b = banana;
  \`\`\`
`;

const CodeBlock: CodeComponent = ({ inline, className, children }) => {
  if (inline) {
    return <code className={className}>{children}</code>;
  }
  const match = /language-(\w+)/.exec(className || "");
  const language = match && match[1] ? match[1] : "";
  return (
    <SyntaxHighlighter style={vscDarkPlus} language={language} PreTag="div">
      {String(children).replace(/\n$/, "")}
    </SyntaxHighlighter>
  );
};

export const MDE: VFC = () => {
  return (
    <div>
      <h1>react-markdown</h1>
      <div>
        <ReactMarkdown
          className="markdown-body"
          components={{
            code: CodeBlock,
            a: ({ href, children }) => (
              <a rel="noreferrer noopener" href={href} target="_blank">
                {children}
              </a>
            ),
          }}
          remarkPlugins={[remarkGfm, breaks]}
        >
          {markdown}
        </ReactMarkdown>
      </div>
    </div>
  );
};
```

## my screen
![image](https://user-images.githubusercontent.com/54790810/137232183-0d930b5f-33b1-4726-ba55-37502f85b453.png)

## What I tried
- markdown.replaceAll(/\\n/gi, '\n &nbsp;')
- markdown.replaceAll(/\\n/gi, '\n \n')

## Issues I checked
- https://github.com/remarkjs/react-markdown/issues/561
- https://github.com/remarkjs/react-markdown/issues/273
- https://github.com/remarkjs/react-markdown/issues/278
- https://github.com/remarkjs/react-markdown/issues/273

# Thank you all.

**Comment by wooorm** [comment, marked as answer]
> Could you please tell me how to get multiple lines on HTML.

HTML does not have “multiple lines”. CSS can be used to add margin and padding between constructs in HTML.
You can use `<br>` elements in HTML, but as with all elements in HTML, it has semantics, aka, a meaning: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/br.

> I’d like to reflect multiple breaks between 'ssssssssssssssss' and ' # h1' because I create a text form which can input both plain text and markdown.

I would recommend that you either allow markdown, or don’t, or let users choose whether they type markdown or plain text

---

A plugin that does somewhat what you want is `remark-breaks`: https://github.com/remarkjs/remark-breaks.

**Reply by hakushun** [reply]
thanks, wooorm!!

>HTML does not have “multiple lines”. CSS can be used to add margin and padding between constructs in HTML.
You can use <br> elements in HTML, but as with all elements in HTML, it has semantics, aka, a meaning: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/br.

I understand what you saying but I am thinking that “multiple lines” realizes in HTML like
```html
<p>
line1



line2
</p>
```

> I would recommend that you either allow markdown, or don’t, or let users choose whether they type markdown or plain text

As you say, this one looks better so add a switch to toggle markdown and plain text.

Really appreciate your help!!!
